### DISCUSSION TITLE
Soft fail does not send the bounce webhook

### DISCUSSION THREAD

**Discussion by sandrovenezuela** [postalserver/postal#2759] [Help with using Postal] [answered]
As you can see in the image below, a Soft Fail is not causing the bounce webhook to be sent.

![image](https://github.com/postalserver/postal/assets/143756648/4976c7bc-a3b9-43c2-b9f0-57c53dea94d8)

After a few attempts, the recipient is added to the suppression list and the error displayed by Postal becomes this: FrozenError: can't modify frozen String: "".

Is there any reason for Postal not to send the bounce webhook in this case? Why does the FrozenError error occur afterwards?

Thanks

**Comment by willpower232** [comment]
The message did not bounce so no bounce webhook will be sent. A bounce would be when the other mail server sends an email back to Postal to say whatever happened to it. The events in your screenshot are happening in real time so the message is never fully received by the other mail server.

As per https://docs.postalserver.io/developer/webhooks, MessageDeliveryFailed will probably be the one on this occasion.

**Comment by sandrovenezuela** [comment]
Hi Will,

But the other server didn't send an email back signaling the problem, in this case, "out of storage space"?

Is there any way to handle this error, perhaps using Routing?

This new example below is different, it shows a Bounce and still no webhook is sent.

![image](https://github.com/postalserver/postal/assets/143756648/7aa3c495-f571-4b59-ae09-bfef61f4015f)

Why?

**Reply by willpower232** [reply]
Each mail server behaves differently, if it knew immediately that there was no storage space available then it didn't need to make the situation worse by trying to store a message it couldn't.

As I said, the MessageDeliveryFailed webhook would have told you there was a problem so you would be able to flag it for further attention.

If webhooks were enabled before the bounced message, it should have sent the bounced webhook. You can have a look at the history to see if it tried

![image](https://github.com/postalserver/postal/assets/1619102/272e7f87-092a-47d9-be8c-cef230c4b638)

**Comment by sandrovenezuela** [comment]
Hi Will,

I have configured both the MessageBounced and MessageDeliveryFailed webhook, and neither of them have been sent.

Postal receives (in Incoming Messages) the Undeliverable message with the Return Path recipient and the only way I found to handle this was using Routing.

Is there a log I can check? Is there any way to track this type of event to understand what is happening?

Thanks!

**Reply by willpower232** [reply]
If you set up the webhooks in the same server as the bounced messages, do you have anything on the history page?

**Reply by sandrovenezuela** [reply]
Hi Will,

No, as I said, Postal never generated a webhook for the MessageBounced event.

I have several accounts configured on Postal, all of them with a Webhook for each event (Bounced and DeliveryFailed) and only the failure event is generated.

What happens in practice is that after several "soft fail" errors, the recipient's email is inserted into the Suppression List.

**Comment by pdifeo** [comment, marked as answer]
> 
> Why does the FrozenError error occur afterwards?
> 

For FrozenError see #2707
